### DISCUSSION TITLE
Can this project be moved to at least "beta"?

### DISCUSSION THREAD

**Discussion by smolinari** [Akryum/vue-apollo#1311] [General]
@Akryum 

Hey Guillaume. Sorry to ping you out of the blue like this, but any chance we can get this package at least to beta? The Quasar app extension adding this package to Quasar is also sticking to beta because this package is still considered alpha. If this package were beta, I'd release our GA version. 

There are almost 40k downloads a week of the composable package now (which is awesome), which means it is being used successfully, I would assume. 

I saw you are working on other OSS projects of yours. Can you put some TLC into this project, as it has a lot of traction. 

Thanks for your consideration in advance.

Scott

**Comment by Akryum** [comment]
Hi! I didn't move it to beta yet as I want to make some API changes. 😉

**Reply by smolinari** [reply]
Awesome. Looking forward to them. Can't wait. 😁 

Scott

**Reply by smolinari** [reply]
@Akryum - Any news? 

Scott

**Comment by betweenbrain** [comment]
Hello @Akryum,

Is there any update on what these API changes may be? Is there anything that our team could do to help implement them?

Thanks!

**Comment by holtergram** [comment]
**Disclaimer**: This here comes as a follow-up on a post I made on Discord in which I said that I'm greatly thankful for all the work that has been put into this and that I honor every line of code. And that I'm well aware that working in the Open Source sphere can be absolutely challenging and that one can only do as much as possible. I don't want to come across unthankful or demanding. It's quite the opposite. As I was writing this I realized that it'd be better to post it here since it's related to this topic.

The thing I would like to get across is that I as a user currently don't know anything about the actual status or timeline. V4 is around for little over 2 years now, if I'm not mistaken. And yet there is no public info on what might change or when it's about to happen. So I'd kindly like to ask why the thoughts and plans of this are not being shared with the community. Which API are going to change? What are the current problems or what's going to be better? s there any ETA on when this project will reach the next (beta) status? 

Communication would make the progress of development much more predictable and transparent for everyone and there are even others who offer their help (like in this very discussion). 

There are still core features like `useLazyQuery` that are not even mentioned in the docs. So right now I actually have to dig in PRs (https://github.com/vuejs/apollo/pull/1309/files) to learn about the package and what it can do – but it doesn't even show me how to use it. So I think that it'd help tremendously to get at least the docs up-to-date.
Furthermore I'd love to read more about the bigger picture like: What are best practices to make Vue-Apollo and Pinia work together (which is still needed for most applications)? Right now it feels like GraphQL is very unsupported (or let's say not featured enough) by Vue which is what I don't understand, because of it's great potential. Even the Discord channel has been retired. Is GraphQL slowly dying in the Vue ecosystem?

Again, I'm absolutely thankful that you (Akryum) took on the whole idea of bringing GraphQL & Apollo to Vue. Many depend on it and use it in their work. I just think that everyone in the community can greatly benefit if there was more communication and open discussions. It would certainly boost the popularity of GraphQL. Just my 2 cents.

**Reply by Akryum** [reply]
`@apollo/options` and `@vue/apollo/components` have stable APIs.
I'd like to make some changes in `@vue/apollo-composable`, here are some of the main ideas:
- Deprecate/remove `useResult` as it can easily be replaced by a normal `computed` (`useResult(result, [], data => data.foo.bar)` would become `computed(() => result?.foo?.bar ?? [])`
- Revise the `useQuery` parameter types and some implementation  to make it simpler
- Rework the progress tracking to work outside of components (or document an alternative)

And obviously improve the documentation 🐈

**Reply by smolinari** [reply]
Sure thing!!! Let's go! Yipeeee!!! :partying_face: 

Scott

**Reply by holtergram** [reply]
> `@apollo/options` and `@vue/apollo/components` have stable APIs.
> 
> I'd like to make some changes in `@vue/apollo-composable`, here are some of the main ideas:
> 
> - Deprecate/remove `useResult` as it can easily be replaced by a normal `computed` (`useResult(result, [], data => data.foo.bar)` would become `computed(() => result?.foo?.bar ?? [])`
> 
> - Revise the `useQuery` parameter types and some implementation  to make it simpler
> 
> - Rework the progress tracking to work outside of components (or document an alternative)
> 
> 
> 
> And obviously improve the documentation 🐈 

We're on beta 🙌🏼 Thanks a lot!

**Reply by smolinari** [reply]
Were the suggested changes made? It doesn't seem like it. But, do correct me if I am wrong. Really glad to see the beta tag. :grinning: 

Scott

**Comment by ebisbe** [comment]
As a long time user of this package I was glad to see it on Beta. At some point I even consider using something else for graphql ( which there are no more realistic options )

**Reply by smolinari** [reply]
Not for Vue with Apollo at least. :) There are other clients though, which do work with Vue. URQL for instance. 

https://formidable.com/open-source/urql/docs/basics/vue/

Scott

**Reply by ebisbe** [reply]
I didn't found that one. I'm using appsync on the backend so I don't really need apollo for that matter.

**Reply by smolinari** [reply]
@ebisbe - That's the cool thing about GraphQL. The client's don't necessarily need a special server (well, except for Relay setups). So, you can use Apollo Client or URQL as you like. It shouldn't affect usage too much. Some have more features than others though. URQL has a good comparison chart. Not sure how up-to-date it is though. 

https://formidable.com/open-source/urql/docs/comparison/

Scott

**Reply by ebisbe** [reply]
That chart is the first think I checked. As of now I don't have the energy to test it but I like to know there's some other options.
